src/pages/Quiz/Quiz.js

```diff
@@ -8,20 +8,36 @@ import head from '../../public/img/RicksHead.png';
  */
 // eslint-disable-next-line no-unused-vars
 function Quiz(props) {
+  const allAnswers = [];
+  const { element: counterElement, increment, getCount } = QuestionCounter();
+  const onClick = () => {
+    // eslint-disable-next-line no-unused-vars
+    const answeredQuestions = getCount() - 1;
+    // eslint-disable-next-line no-unused-vars
+    const points = allAnswers.filter((answer) => answer.correct).length;
+  };
   // @ts-ignore
-  const { question } = Question(props.generator.next().value);
+  let question = Question(props.generator.next().value);
```

Comment:
Rozwiązanie problemu z calla:

System typizacji obawia się, że generator nie jest nieskończony (bo nie wie że jest) i że kiedyś zabraknie następnego pytania.

Żeby to rozwiązać, w `src/pages/Loading/Loading.js` poprawić definicję typu `QuestionGenerator`. Teraz jest to (skracając importy):
```TS
Generator<CharacterQuestion | EpisodeOrLocationQuestion, void, unknown>
```

Gdzie pierwsza zmienna typowa (`CharacterQuestion | EpisodeOrLocationQuestion`) to wartość `yield`owana z generatora, a druga (`void`), `return`owana ([źródło](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-6.html#stricter-generators)). Obecnie TS myśli, że z funkcja może się skończyć (bez zwracania niczego), a możemy zaznaczyć, że wiemy, że nigdy nie zwraca: `never`. Czyli skończy się na:
```TS
Generator<CharacterQuestion | EpisodeOrLocationQuestion, never, unknown>
```

Potem trzeba powiedzieć, że nasz generator faktycznie ma ten typ, czyli w `src/data/questions.js` dać dobry typ zwracany:
```TS
@returns {import('../pages/Loading').QuestionGenerator}
```

I z nie wiem jakiego powodu TS nie widzi `while (1)` jako nieskończoną pętlę, a `while (true)` już tak, więc trzeba też to zmienić 2 linijki niżej. I wtedy można usunąć `// @ts-ignore` bez błędów.

**TL;DR**, bo to spory komentarz jak na kilka linii zmian:

1. Linię 16 w `Loading.js` zmienić na:
```TS
 * @typedef {Generator<import('../../data/questions').CharacterQuestion | import('../../data/questions').EpisodeOrLocationQuestion, never, unknown>} QuestionGenerator
```
2. W `questions.js` dodać w linii 123:
```TS
 * @returns {import('../pages/Loading').QuestionGenerator}
```

3. W `questions.js` w linii 126 zmienić `while (1)` na `while (true)`.

4. Tutaj usunąć `// @ts-ignore`.

(w sumie chyba najfajniej byłoby przenieść definicję `QuestionGenerator` do `questions.js` z `loading.js`)